src/proxy/MultiMapProxy.ts

```diff
@@ -0,0 +1,204 @@
+import * as Promise from 'bluebird';
+
+import {MultiMap} from './MultiMap';
+import {BaseProxy} from './BaseProxy';
+import {IMapListener} from '../core/MapListener';
+import {MultiMapPutCodec} from './../codec/MultiMapPutCodec';
+import {MultiMapGetCodec} from './../codec/MultiMapGetCodec';
+import {Data} from '../serialization/Data';
+import {MultiMapRemoveCodec} from './../codec/MultiMapRemoveCodec';
+import {MultiMapRemoveEntryCodec} from './../codec/MultiMapRemoveEntryCodec';
+import {MultiMapKeySetCodec} from './../codec/MultiMapKeySetCodec';
+import {MultiMapValuesCodec} from './../codec/MultiMapValuesCodec';
+import {MultiMapEntrySetCodec} from './../codec/MultiMapEntrySetCodec';
+import {MultiMapContainsKeyCodec} from './../codec/MultiMapContainsKeyCodec';
+import {MultiMapContainsValueCodec} from './../codec/MultiMapContainsValueCodec';
+import {MultiMapContainsEntryCodec} from './../codec/MultiMapContainsEntryCodec';
+import {MultiMapSizeCodec} from './../codec/MultiMapSizeCodec';
+import {MultiMapClearCodec} from './../codec/MultiMapClearCodec';
+import {MultiMapValueCountCodec} from './../codec/MultiMapValueCountCodec';
+import {EntryEventType} from '../core/EntryEventType';
+import ClientMessage = require('../ClientMessage');
+import {MultiMapAddEntryListenerToKeyCodec} from './../codec/MultiMapAddEntryListenerToKeyCodec';
+import {MultiMapAddEntryListenerCodec} from './../codec/MultiMapAddEntryListenerCodec';
+import {MultiMapRemoveEntryListenerCodec} from './../codec/MultiMapRemoveEntryListenerCodec';
+import {MultiMapLockCodec} from '../codec/MultiMapLockCodec';
+import {MultiMapIsLockedCodec} from '../codec/MultiMapIsLockedCodec';
+import {MultiMapTryLockCodec} from '../codec/MultiMapTryLockCodec';
+import {MultiMapUnlockCodec} from '../codec/MultiMapUnlockCodec';
+import {MultiMapForceUnlockCodec} from '../codec/MultiMapForceUnlockCodec';
+
+export class MultiMapProxy<K, V> extends BaseProxy implements MultiMap<K, V> {
+
+    put(key: K, value: V): Promise<boolean> {
+        var keyData = this.toData(key);
+        var valueData = this.toData(value);
+        return this.encodeInvokeOnKey<boolean>(MultiMapPutCodec, keyData, keyData, valueData, 1);
+    }
+
+    get(key: K): Promise<Array<V>> {
+        var keyData = this.toData(key);
+
+        var deserializeCallback = (items: Array<Data>) => {
+            return items.map((item) => {
```

Comment:
No need to create second function here. This can be replaced with simply `return items.map(this.toObject)`